### PR TITLE
ShyLU FastILU: simple optimizations

### PR DESCRIPTION
**Pull request by brian-kelley**
Remove unneeded Kokkos fences, view initializations. To make sure kernels are still run in proper sequence without the fences, specify the execution space on every RangePolicy (instead of using default space). Also replace a sort_crs_matrix with sort_crs_graph, since in symbolic the matrix values haven't been initialized yet.

Still keep the fences after the top-level functions that users can call directly (initialize, compute, apply) so that initTime/computeTime/applyTime remain accurate. Also keep fences before other timing measurements (but these are guarded by ``#ifdef FASTILU_TIMER``).

@trilinos/shylu 

## Motivation

Performance improvements

## Stakeholder Feedback

This could benefit performance on Sierra milestone problems.
## Testing

Tested by running from the Ifpack2+Belos driver: RCM + FastILU + Host trsv, and METIS + FastILU + FastSptrsv. Ran on Cuda, Cuda+UVM, and Serial. The Serial ran clean under Valgrind, so no uninitialized memory issues.

### CODE REVIEW

**Review by iyamazaki — APPROVED** [review]
Thank you, @brian-kelley. These changes look good to me.

**Comment by brian-kelley** [comment]
@srajama1 @ndellingwood This needs a review by a ShyLU code owner, could one of you please look at it? Thanks!

**Comment by jennloe** [comment]
Or perhaps @egboman @ambrad or @kyungjoo-kim could leave a ShyLu review?

**Pull request merged**
